prime-router/src/main/kotlin/docgenerators/DocumentationFactory.kt

```diff
@@ -0,0 +1,80 @@
+package gov.cdc.prime.router.docgenerators
+
+import gov.cdc.prime.router.Schema
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+/** The base documentation factory with core logic */
+abstract class DocumentationFactory {
+    abstract val fileExtension: String
+
+    abstract fun getSchemaDocumentation(schema: Schema): Sequence<*>
```

Comment:
Using the `*` means that you can pass a sequence of unknown type in subclasses. This is different than using `Any` which is like using `object`